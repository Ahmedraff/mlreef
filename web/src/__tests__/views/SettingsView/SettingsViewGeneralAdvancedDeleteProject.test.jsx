import React from 'react';
import { mount } from 'enzyme';
import renderer from 'react-test-renderer';
import DeleteProject from 'components/views/SettingsView/SettingsViewGeneralAdvancedDeleteProject';

const onDelete = () => 'deleted';

describe('basic behaviour', () => {
  test('assert that snapshot matches', () => {
    const component = (
      <DeleteProject name="Awesome project Name" onDelete={onDelete} />
    );

    const snapShot = renderer.create(component).toJSON();
    expect(snapShot).toMatchSnapshot();
  });

  test('assert that confirm button is disabled', () => {
    const wrapper = mount(
      <DeleteProject name="Awesome project Name" onDelete={onDelete} />,
    );

    const isDisabled = wrapper.find('button')
      .getElement()
      .props.disabled;

    expect(isDisabled).toBe(true);
  });
});
