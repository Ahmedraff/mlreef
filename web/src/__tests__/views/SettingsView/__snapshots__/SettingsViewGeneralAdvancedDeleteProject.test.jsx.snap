// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`basic behaviour assert that snapshot matches 1`] = `
<div>
  <p>
    <b
      className="t-danger"
    >
      You are going to remove 
      Awesome project Name
      . Removed project CANNOT be restored! Are you ABSOLUTELY sure?
    </b>
  </p>
  <p>
    This action can lead to data loss. To prevent accidental actions we ask your permission to confirm your intention.
    Please type 
    <span
      className="bg-light t-danger"
    >
      Awesome project Name
    </span>
     to proceed or close this modal.
  </p>
  <input
    className="mb-4"
    data-cy="project-name-input"
    onChange={[Function]}
    value=""
  />
  <button
    className="btn btn-danger"
    data-cy="remove-project"
    disabled={true}
    onClick={[Function]}
    type="button"
  >
    Confirm
  </button>
</div>
`;
